glotaran/analysis/problem.py

```diff
@@ -323,7 +323,8 @@ def create_result_data(
     ) -> dict[str, xr.Dataset]:
 
         if history_index is not None and history_index != -1:
-            self.parameters = self.parameter_history[history_index]
+            valid_idx = -2 if len(self.parameter_history) > 1 else -1
+            self.parameters = self.parameter_history[valid_idx]
```

Comment:
Can crash if only a single item has been added to parameter_history, in any case -2 might not be relevant anymore.

Reply:
so the thing is: the intial are always the zeroth element. If they crash then there are no sane parameters to work with and a result cennot be created, You need to raise Exception then. But this must be done in `optimize.py`

The paramters are saved at begin of an function evaluation. After the opimization, the last parameters are saved (the last evaluation, not the optimal ones!), which are the one at which it crashed if not successful. So when you not succed, you need to get -2th element. Since in case of crash with initial, the initial will be saved twice (at start of iter and after recover from crash) it is guranteed that we always have 2 elements in the history.